cabby/geo/map_processing/map_structure.py

```diff
@@ -69,7 +69,9 @@ def __init__(
       self.add_poi_to_graph()
       self.nodes, self.edges = ox.graph_to_gdfs(self.nx_graph)
 
-
+      osmid = self.nx_graph.nodes
+      osmid_str = list(map(str, osmid))
+      nx.set_edge_attributes(self.nx_graph, osmid_str, 'osmid')
```

Comment:
This will create an `osmid` attribute for every node with the value equal to the _entire_ `osmid_str` list. Looking at the [docs for set_node_attributes](https://networkx.org/documentation/networkx-2.0/reference/generated/networkx.classes.function.set_node_attributes.html), I think you want `osmid_str` to be a dictionary mapping the node name to the string version of the osmid.